specifications/sequence-diagrams: add withdraw seq diagram

acharb opened the pull request:
adds a sequence diagram going through the [withdrawal steps](https://github.com/stellar/experimental-payment-channels/blob/main/specifications/sep-payment-channel-mechanism.md#withdraw) of the payment channel SEP

acharb commented:
see diagram [here](https://mermaid-js.github.io/mermaid-live-editor/#/edit/eyJjb2RlIjoic2VxdWVuY2VEaWFncmFtXG4gICAgcGFydGljaXBhbnQgSSBhcyBJbml0aWF0b3JcbiAgICBwYXJ0aWNpcGFudCBSIGFzIFJlc3BvbmRlclxuXG4gICAgTm90ZSBvdmVyIEksUjogMS4gU2V0dXAgYW5kIGEgc2luZ2xlIHVwZGF0ZVxuICAgIE5vdGUgcmlnaHQgb2YgUjogaSA9IDIsIGUgPSAwXG4gICAgTm90ZSBsZWZ0IG9mIEk6IEVJLnNlcU51bSA9IDEwMDBcblxuICAgIE5vdGUgb3ZlciBJLFI6IDIuIFdpdGhkcmF3YWwgYnkgSVxuICAgIEktPj5JOiBpbmNyZW1lbnQgaVxuICAgIFItPj5SOiBpbmNyZW1lbnQgaVxuICAgIE5vdGUgcmlnaHQgb2YgUjogaSA9IDNcbiAgICBJLT4-UjogY3JlYXRlIFdfMywgd2hpY2ggaGFzIGEgcGF5bWVudCBvcCBmcm9tIEVJIHRvIEksIGFuZCBidW1wcyBFSS5zZXFOdW0gdG8gc18zXG4gICAgSS0-Pkk6IHNldCBlJyA9IGUsIHNldCBlID0gaSwgdGhlbiBpbmNyZW1lbnQgaVxuICAgIFItPj5SOiBzZXQgZScgPSBlLCBzZXQgZSA9IGksIHRoZW4gaW5jcmVtZW50IGlcbiAgICBOb3RlIHJpZ2h0IG9mIFI6IGUnID0gMCwgZSA9IDMsIGkgPSA0XG4gICAgSS0-PlI6IGNyZWF0ZSBhbmQgc2lnbiBDXzQsIHdoaWNoIGhhcyB0aGUgbGFzdCBhZ3JlZWQgdXBvbiBwYXlvdXRzIG1pbnVzIEkncyBwYXlvdXQgZnJvbSBXXzNcbiAgICBSLS0-Pkk6IHNpZ24gQ180LCBzZW5kIGJhY2tcbiAgICBJLT4-UjogY3JlYXRlIGFuZCBzaWduIERfNFxuICAgIFItLT4-STogc2lnbiBEXzQsIHNlbmQgYmFja1xuICAgIFItPj5JOiBzaWduIFdfNFxuICAgIEktPj5JOiBzdWJtaXQgV180XG5cbiAgICBhbHQgV180IHN1Y2Nlc3NmdWxcbiAgICAgICAgTm90ZSByaWdodCBvZiBSOiB3aXRoZHJhd2FsIGNvbXBsZXRlXG4gICAgICAgIE5vdGUgbGVmdCBvZiBJOiBFSS5zZXFOdW0gPSAxMDA3XG4gICAgZWxzZSBXXzQgdW5zdWNjZXNzZnVsXG4gICAgICAgIEktPj5JOiBzZXQgZSA9IGUnXG4gICAgICAgIFItPj5SOiBzZXQgZSA9IGUnXG4gICAgICAgIE5vdGUgcmlnaHQgb2YgUjogZSA9IDBcbiAgICBlbmQiLCJtZXJtYWlkIjp7fSwidXBkYXRlRWRpdG9yIjpmYWxzZX0)

leighmcculloch reviewed (approved):
Looks right to me, easy to follow in the diagram. 👏🏻 

I think in the withdraw case it's also worthwhile showing what a following update would look like in the successful and unsuccessful case, and demonstrating why C_4/D_4 are useless and can't be used. This might be too hard to show in a diagram though. Thoughts?

acharb commented:
@leighmcculloch ya I tried adding the C_4 and D_4 construction notes to it but it was too complicated so I removed it. I added a note about D_4 at the end [here](https://github.com/stellar/experimental-payment-channels/pull/16/commits/091dec805ea064a1856f604d3e47f845c6e0fdcd), imo this is enough info w/o being too distracting
